src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/KafkaInstrumentation.cs

```diff
@@ -116,6 +117,16 @@ public static void SetDeliveryResults(Activity activity, IDeliveryResult deliver
             deliveryResult.Offset.Value);
     }
 
+    internal static string? ExtractMessageKeyValue(object key)
+    {
+        return key switch
+        {
+            string s => s,
+            int or long or float or double => Convert.ToString(key, CultureInfo.InvariantCulture),
```

Comment:
I suggest also adding:
- `uint`
- `ulong`
- `decimal`

(even if they are not yet handled by the instrumented library

Reply:
byte, short, etc ; )